public/app/features/dashboard-scene/settings/variables/utils.ts

```diff
@@ -1,14 +1,102 @@
-import { DashboardScene } from '../../scene/DashboardScene';
-
-export function getVariableTypes(dashboard: DashboardScene) {
-  // TODO: this should get the existing variable types from grafana scenes
-
-  return [
-    { value: 'constant', label: 'Constant' },
-    { value: 'query', label: 'Query' },
-    { value: 'interval', label: 'Interval' },
-    { value: 'datasource', label: 'Datasource' },
-    { value: 'textbox', label: 'Text Box' },
-    { value: 'custom', label: 'Custom' },
-  ];
+import { SelectableValue } from '@grafana/data';
+import {
+  AdHocFiltersVariable,
+  ConstantVariable,
+  CustomVariable,
+  DataSourceVariable,
+  IntervalVariable,
+  TextBoxVariable,
+  QueryVariable,
+} from '@grafana/scenes';
+import { VariableType } from '@grafana/schema';
+
+import { AdHocFiltersVariableEditor } from './editors/AdHocFiltersVariableEditor';
+import { ConstantVariableEditor } from './editors/ConstantVariableEditor';
+import { CustomVariableEditor } from './editors/CustomVariableEditor';
+import { DataSourceVariableEditor } from './editors/DataSourceVariableEditor';
+import { IntervalVariableEditor } from './editors/IntervalVariableEditor';
+import { QueryVariableEditor } from './editors/QueryVariableEditor';
+import { TextBoxVariableEditor } from './editors/TextBoxVariableEditor';
+
+interface EditableVariableConfig {
+  name: string;
+  description: string;
+  // FIXME: This should include all the scene objects
+  scene:
+    | typeof CustomVariable
+    | typeof QueryVariable
+    | typeof ConstantVariable
+    | typeof IntervalVariable
+    | typeof DataSourceVariable
+    | typeof AdHocFiltersVariable
+    | typeof TextBoxVariable;
+  // FIXME: This should include all the editor objects
+  editor: React.ComponentType<any>;
+}
+
+export type EditableVariableType = Exclude<VariableType, 'system'>;
+
+export function isEditableVariableType(type: VariableType): type is EditableVariableType {
+  return type !== 'system';
+}
+
+const EDITABLE_VARIABLES: Record<EditableVariableType, EditableVariableConfig> = {
```

Comment:
This is the "registry" where all the definitions exist. I typed in a way you must add a new one here in case it is added to `VariableType`.